### PR TITLE
feature(planner): Introduce `InterpreterFactoryV2` for new planner

### PR DESCRIPTION
**Pull request by leiysky**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Changelog

- New Feature

### CODE REVIEW

File: query/src/sql/planner/mod.rs [hunk 1]
```diff
@@ -54,13 +54,13 @@ impl Planner {
         Planner { ctx }
     }
 
-    pub async fn plan_sql(&mut self, sql: &str) -> Result<(NewPipeline, Vec<NewPipeline>)> {
-        let (optimized_expr, bind_context, metadata) = self.build_sexpr(sql).await?;
-        self.build_pipeline(optimized_expr, bind_context, metadata)
-            .await
-    }
+    // pub async fn plan_sql(&mut self, sql: &str) -> Result<(NewPipeline, Vec<NewPipeline>)> {
```
Comment (xudong963) [inline query/src/sql/planner/mod.rs:57]: Forget to delete it? :🤔

Comment (leiysky) [reply query/src/sql/planner/mod.rs]: Fixed

File: query/src/sql/planner/mod.rs [hunk 2]
```diff
@@ -73,24 +64,12 @@ impl Planner {
...
-        let bind_result = binder.bind(&stmts[0]).await?;
+        let plan = binder.bind(&stmts[0]).await?;
 
         // Step 3: optimize the SExpr with optimizers, and generate optimized physical SExpr
-        let optimize_context = OptimizeContext::create_with_bind_context(&bind_result.bind_context);
-        let optimized_expr = optimize(bind_result.s_expr, optimize_context)?;
-
-        Ok((optimized_expr, bind_result.bind_context, metadata.clone()))
-    }
+        // let optimize_context = OptimizeContext::create_with_bind_context(&bind_result.bind_context);
```
Comment (xudong963) [inline query/src/sql/planner/mod.rs:70]: ditto

**Review by xudong963 — APPROVED** [review]
LGTM 👍

File: query/src/sql/planner/mod.rs (hunk 2 above)
Comment (leiysky) [reply query/src/sql/planner/mod.rs]: Fixed.

**Pull request merged**